components/AppLayout/AppHeader.js

```diff
@@ -1,66 +1,244 @@
+
+  const openProfileMenu = useCallback(e => setAnchor(e.currentTarget), [
+    anchor,
+  ]);
+  const closeProfileMenu = useCallback(() => setAnchor(null), [anchor]);
+
+  return (
+    <header className={classes.root}>
+      <div className={classes.top}>
+        <div className={classes.flex}>
+          <a href="/">
+            <img
+              className={classes.logo}
+              src={mobileLogo}
+              srcSet={`${desktopLogo} ${theme.breakpoints.values.md}w`}
+              alt=""
+            />
+          </a>
+          <Box display={['none', 'none', 'inline']}>
+            <Links classes={classes} />
+          </Box>
+        </div>
+        {/* GlobalSearch not fully implemented yet */}
+        {/* <GlobalSearch />*/}
+        <Box display={['none', 'none', 'block']}>
+          {user?.name ? (
+            <>
+              <Widgets.Avatar user={user} size={40} onClick={openProfileMenu} />
+              <Menu
+                id="profile-menu"
+                classes={{ paper: classes.profileMenu }}
+                anchorEl={anchor}
+                keepMounted
+                open={!!anchor}
+                onClose={closeProfileMenu}
+              >
+                <Widgets.Level user={user} />
+                <MenuItem onClick={closeProfileMenu}>
+                  <ListItemIcon>
+                    <Widgets.Avatar user={user} size={40} />
+                  </ListItemIcon>
+                  <Typography variant="inherit">{user?.name}</Typography>
+                </MenuItem>
+                <Divider classes={{ root: classes.divider }} />
+                <MenuItem onClick={closeProfileMenu}>
+                  <ListItemIcon className={classes.listIcon}>
+                    <AccountCircleOutlinedIcon />
+                  </ListItemIcon>
+                  <Typography variant="inherit">{t`My Profile`}</Typography>
+                </MenuItem>
+                <Divider classes={{ root: classes.divider }} />
+                <MenuItem onClick={closeProfileMenu}>
+                  <ListItemIcon className={classes.listIcon}>
+                    <InfoIcon />
+                  </ListItemIcon>
+                  <Typography variant="inherit">{t`About`}</Typography>
+                </MenuItem>
+                <Divider classes={{ root: classes.divider }} />
+                <MenuItem onClick={logout}>
+                  <ListItemIcon className={classes.listIcon}>
+                    <ExitToAppRoundedIcon />
+                  </ListItemIcon>
+                  <Typography variant="inherit">{t`Logout`}</Typography>
+                </MenuItem>
+              </Menu>
+            </>
+          ) : (
+            <Button
+              onClick={openLoginModal}
+              size="small"
+              className={classes.login}
+            >{t`Login`}</Button>
+          )}
+        </Box>
+      </div>
+      <Box display={['block', 'block', 'none']}>
+        <div className={classes.flex}>
```

Comment:
Can use `<Box display={['flex', 'flex', 'none']}></Box>` instead of 2 nested `div`s, simplify the DOM